File: backend/database/website_info.py
from uuid import UUID, uuid4

from sqlmodel import Field, SQLModel


class WebsiteInfo(SQLModel, table=True):
    id: UUID = Field(default_factory=uuid4, primary_key=True)
    logo: str
    name: str = Field(max_length=36)
    description: str = Field(max_length=64)
    icp: str | None = Field(default=None, max_length=128)

File: frontend/src/types/website-info.ts
export type WebsiteInfoResponseSchemas = {
  id: string;
  logo: string;
  name: string;
  description: string;
  icp: string | null;
};

export type UpdateWebsiteInfoRequestSchemas = {
  name: string;
  description: string;
  icp: string | null;
};
